backend/repository/config.go

```diff
@@ -378,10 +378,31 @@ func (r *Repository) endpointConfigs(thriftRootDir string, gatewaySpec *codegen.
 func (r *Repository) middlewareConfigs(
 	gatewaySpec *codegen.GatewaySpec,
 ) (map[string]interface{}, map[string]*RawMiddlewareConfig, error) {
-	middlewareSpecs := gatewaySpec.MiddlewareModules
+	orgMiddlewareSpecs := gatewaySpec.MiddlewareModules
```

Comment:
any specific reason we're not doing this in `parseMiddlewareConfig`? it seems to be one single place parsing middlewares, files or not.

Reply:
thought about that.  mostly, I was worried that putting them on GatewaySpec would mess something up with the module system  since I think the middleware handler codegen runs off of that list,  also  I think permeating the special casing deeper into the core  just adds complexity.     Pretty easy to revisit if we look at it more.

Reply:
how we create middleware specs (should have all information with all types ... )
and how middleware specs are used, like in here:https://github.com/uber/zanzibar/blob/master/codegen/gateway.go#L645
(it's literally check `transformRequest` and skip it)
is separate concerns.

If middleware consumer mess up we should fix the consumer.

Anyway just raised the concern that middleware should be generated in one place

Reply:
the way transforms are implemented was a hack into middleware  i would be more interested in the idea of moving x-form spec and generation to Endpoint.   For the most part this is trying to quickly fix a bug between a few large incompatible subsystems Codegen, Orchestrater API and Module System.

Reply:
👍